### ISSUE TITLE
Waku v2: Implement rpc message flow for WakuFilter

### ISSUE THREAD

**Issue by kdeme** [status-im/nim-waku#167]
WakuFilter protocol currently only sends data here: https://github.com/status-im/nim-waku/blob/master/waku/protocol/v2/waku_filter.nim#L125

There is no way to trigger e.g. a handler or so on the receival of data. (e.g. there should be some sort of reading loop which triggers some handler(s), e.g. the filter handlers, to then forward the messages).
Additionally, it will require the selection of the right message (as there are two, FilterRequest and MessagePush), and its responses. 

To be able to make this selection, the protocol will need either:
 - Some additional identifier before the actual message
 - Or one could use a different libp2p protocol identifier per type of message.

For responses, one might also need a specific request id.

So basically, the whole message dispatching still needs to be implemented, unless I'm missing something and libp2p foresees code for this. You can however let libp2p do a chunk of the work by using different libp2p protocol identifiers per message (afaik this is the way it is often done in libp2p? @dryajov ?)

**Comment by kdeme** [comment]
This was also pointed out here actually: https://github.com/status-im/nim-waku/pull/160#issuecomment-692943594

**Comment by oskarth** [comment]
@decanus is this clear and are you planning on doing this?

**Comment by oskarth** [comment]
Can we make the acceptance criteria of this more precise?

**Comment by decanus** [comment]
https://github.com/status-im/nim-waku/blob/master/waku/protocol/v2/waku_filter.nim#L101-L112 This is the way it works now

**Comment by oskarth** [comment]
@kdeme are you happy with this solution?
